workflows/flows/assemble_study_tasks/archive_assembly_dirs.py

```diff
@@ -0,0 +1,92 @@
+import operator
+from functools import reduce
+from pathlib import Path
+
+from prefect import flow, get_run_logger
+
+from activate_django_first import EMG_CONFIG
+from analyses.models import Study, Assembly
+
+
+@flow(
+    name="Archive assembly directories",
+    flow_run_name="Archive assembly directories of {reads_study_mgys}",
+)
+def archive_assembly_dirs(reads_study_mgys: str, dry_run: bool = True):
+    """
+    Currently this deletes miassembler workdirs provided the study is in a suitable state of uploaded/inactive assemblies.
+    In future, it might move assembly dirs to LTS or a results archive if needed.
+    """
+    logger = get_run_logger()
+
+    study = Study.objects.get(accession=reads_study_mgys)
+    logger.info(f"Study is {study}")
+
+    assemblies_started = study.assemblies_reads.filter_by_statuses(
+        [Assembly.AssemblyStates.ASSEMBLY_STARTED]
+    )
+    logger.info(
+        f"Study has {assemblies_started.count()} reads-assemblies in STARTED state"
+    )
+
+    is_terminal_filters = study.assemblies_reads.get_queryset()._build_q_objects(
+        keys=[
+            Assembly.AssemblyStates.ASSEMBLY_UPLOADED,
+            Assembly.AssemblyStates.ASSEMBLY_BLOCKED,
+            Assembly.AssemblyStates.ASSEMBLY_FAILED,
+            Assembly.AssemblyStates.PRE_ASSEMBLY_QC_FAILED,
+            Assembly.AssemblyStates.POST_ASSEMBLY_QC_FAILED,
+        ],
+        allow_null=True,
+        truthy_target=True,
+    )
+
+    is_terminal = reduce(operator.or_, is_terminal_filters)
+
+    if (started_not_finished := assemblies_started.exclude(is_terminal)).exists():
+        logger.warning(
+            f"{started_not_finished.count()} of the assemblies are not in any terminal state."
+        )
+        logger.warning(f"**Not** cleaning {study}")
+        return
+
+    # Assembly workdir roots could be in any ena-accession prefixed dir, with or without samplesheet hash
+    # e.g. one of
+    # /nfs/.../slurm_workdir/ERP123_miassembler/
+    # /nfs/.../slurm_workdir/PRJ123_miassembler/
+    # /nfs/.../slurm_workdir/ERP123_miassembler/abcdef123456
+    # /nfs/.../slurm_workdir/PRJ123_miassembler/abcdef123456
+    # Look directly, since we may be cleaning folders that have not appeared in an assembly .dir field (if all FAILED)
+
+    potential_workdirs_roots = {
+        Path(EMG_CONFIG.slurm.default_workdir)
+        / f"{study.ena_study.accession}_miassembler",
+    }.union(
+        {
+            Path(EMG_CONFIG.slurm.default_workdir) / f"{acc}_miassembler"
+            for acc in study.ena_accessions
+        }
+    )
+
+    for potential_workdirs_root in potential_workdirs_roots:
+        if not potential_workdirs_root.is_dir():
+            logger.info(
+                f"No directory found at {potential_workdirs_root}. Nothing to be done."
+            )
+        logger.info(f"Looking for workdirs under {potential_workdirs_root}")
+
+        if (workdir := potential_workdirs_root / "work").is_dir():
```

Comment:
Does this need to be an else? I assume if potential_workdirs_root is not a directory then it doesn't need to check folders within it